src/Data/DataSource.php

```diff
@@ -677,29 +675,28 @@ public static function resolve_node( $global_id, AppContext $context, ResolveInf
 		$id_components = Relay::fromGlobalId( $global_id );
 
 		/**
-		 * If the $id_components is a proper array with a type and id
+		 * $id_components is an array with the id and type
 		 *
 		 * @since 0.0.5
 		 */
-		if ( is_array( $id_components ) && ! empty( $id_components['id'] ) && ! empty( $id_components['type'] ) ) {
-
-			/**
-			 * Get the allowed_post_types and allowed_taxonomies
-			 *
-			 * @since 0.0.5
-			 */
+		if ( empty( $id_components['id'] ) || empty( $id_components['type'] ) ) {
```

Comment:
Early return to make it more obvious what this was returning. Functionality is identical.